mp/src/game/client/momentum/ui/leaderboards/LeaderboardsTimes.cpp

```diff
@@ -365,27 +383,37 @@ void CLeaderboardsTimes::LoadLocalTimes(KeyValues* kv)
             char pReplayPath[MAX_PATH];
             V_ComposeFileName(RECORDING_PATH, pFoundFile, pReplayPath, MAX_PATH);
 
-            CMomReplayBase *pBase = g_ReplayFactory.LoadReplayFile(pReplayPath, false);
-            Assert(pBase != nullptr);
-
-            if (pBase)
-                m_vLocalTimes.InsertNoSort(pBase);
+            FileAsyncRequest_t request;
+            request.pszFilename = pReplayPath;
+            request.pContext = this;
+            request.pfnCallback = LocalTimesCallbackForwarder;
+            request.pszPathID = "MOD";
+            filesystem->AsyncRead(request);
 
             pFoundFile = filesystem->FindNext(found);
         }
 
         filesystem->FindClose(found);
 
-        if (!m_vLocalTimes.IsEmpty())
-        {
-            m_vLocalTimes.RedoSort();
-            m_bTimesNeedUpdate[TIMES_LOCAL] = false;
-        }
+        m_bTimesNeedUpdate[TIMES_LOCAL] = false;
+        m_flTimesLastUpdate[TIMES_LOCAL] = gpGlobals->curtime;
     }
+}
+
+void CLeaderboardsTimes::LocalTimesCallback(const FileAsyncRequest_t &request, int nBytesRead, FSAsyncStatus_t err)
+{
+    if(!request.pData || err != FSASYNC_OK)
+        return;
+
+    CUtlBuffer reader;
+    reader.SetExternalBuffer(request.pData, nBytesRead, nBytesRead, CUtlBuffer::READ_ONLY);
 
-    // Convert
-    if (!m_vLocalTimes.IsEmpty())
-        ConvertLocalTimes(kv);
+    auto replay = g_ReplayFactory.LoadReplayFromBuffer(reader, false);
+    if(replay)
+    {
+        m_vLocalTimes.Insert(replay);
+        PostMessage(GetVParent(), new KeyValues("LocalTimeProcessed"));
```

Comment:
Should it be to the parent or this panel?

Reply:
It is supposed to be the parent, so it can call Update and deal with the update timer on itself but it could also be the panel itself and only call FillLeaderboards, IMO the first one is cleaner

Reply:
Ah gotcha, all good then 👍